catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,715 @@
+        when(preQueryPlugin.process(any(),
+                any())).thenAnswer(invocation -> invocation.getArguments()[1]);
+
+        clientAdaptor = mock(SolrClientAdaptor.class);
+
+        cache = mock(SolrCache.class);
+
+        validationQueryFactory = mock(ValidationQueryFactory.class);
+
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(preQueryPlugin),
+                new ArrayList<>(),
+                cache,
+                cacheExecutor,
+                validationQueryFactory,
+                new CacheQueryFactory(new GeotoolsFilterBuilder()));
+
+        mockSortedQueryMonitorFactory = mock(SortedQueryMonitorFactory.class);
+        mockSortedQueryMonitor = mock(SortedQueryMonitor.class);
+
+        responseArgumentCaptor = ArgumentCaptor.forClass(QueryResponseImpl.class);
+        requestArgumentCaptor = ArgumentCaptor.forClass(QueryRequestImpl.class);
+
+        when(mockSortedQueryMonitorFactory.createMonitor(any(CompletionService.class),
+                any(Map.class),
+                responseArgumentCaptor.capture(),
+                requestArgumentCaptor.capture(),
+                any(List.class))).thenReturn(mockSortedQueryMonitor);
+
+        strategy.setSortedQueryMonitorFactory(mockSortedQueryMonitorFactory);
+
+        mockQuery = mock(Query.class);
+        when(mockQuery.getTimeoutMillis()).thenReturn(LONG_TIMEOUT);
+        when(mockQuery.getPageSize()).thenReturn(-1);
+
+        mockMetacard = new MetacardImpl();
+
+        mockResult = new ResultImpl(mockMetacard);
+
+        mockResponse = mock(SourceResponse.class);
+        results = Arrays.asList(mockResult);
+        when(mockResponse.getResults()).thenReturn(results);
+
+        // Set general properties
+        properties = new HashMap<>();
+        properties.put(QUERY_MODE, NATIVE_QUERY_MODE);
+
+    }
+
+    @After
+    public void tearDownClass() throws Exception {
+        strategy.shutdown();
+    }
+
+    @Test
+    public void testFederateQueryCache() throws Exception {
+        properties.put(QUERY_MODE, CACHE_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        Source mockSource = mock(Source.class);
+        when(mockSource.query(any(QueryRequest.class))).thenReturn(mockResponse);
+
+        QueryResponse federateResponse = strategy.federate(Arrays.asList(mockSource),
+                fedQueryRequest);
+
+        assertThat(requestArgumentCaptor.getValue()
+                .getPropertyValue(QUERY_MODE), is(CACHE_QUERY_MODE));
+
+        verify(cache).query(any(QueryRequest.class));
+        verify(mockSource, times(0)).query(any(QueryRequest.class));
+
+        assertThat(federateResponse.getRequest()
+                        .getQuery(),
+                is(requestArgumentCaptor.getValue()
+                        .getQuery()));
+    }
+
+    @Test
+    public void testFederateQueryCacheNoUpdate() throws Exception {
```

Comment:
I would have expected this test and the next two to verify if the cache is or is not updated as expected based on the query mode.